src/java/htsjdk/tribble/TabixFeatureReader.java

```diff
@@ -126,7 +128,8 @@ public boolean hasIndex(){
     }
 
     public CloseableTribbleIterator<T> iterator() throws IOException {
-        final InputStream is = new BlockCompressedInputStream(ParsingUtils.openInputStream(path));
+    	final InputStream compressedStream = ParsingUtils.openInputStream(path);
+        final InputStream is = Defaults.USE_ASYNC_IO_FOR_TRIBBLE ? new AsyncBlockCompressedInputStream(compressedStream) : new BlockCompressedInputStream(compressedStream); 
```

Comment:
Can we limit this PR to BAMs (as per title)? Tribble is a whole new can of worms and let's keep it as is for now.